projects/common/lib/components/date/date.component.ts

```diff
@@ -263,6 +263,11 @@ You must use either [restrictDate] or the [dateRange*] inputs.
       this.dateChange.emit(this.date);
     } else {
       this.destroyDate();
+
+      // Trigger validator for emptying fields use case. This is to remove the 'Invalid date' error.
```

Comment:
Suggestion, move this logic for empty fields into the destroyDate method.  That way if this method is call somewhere else we do not encounter the invalid error not being cleared.